### PR TITLE
sage: add libpng to sagelib, import package update patches

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Since the 3.7.0 upgrade (that is, since the last staging-next merge #219444), matplotlib no longer propagates `libpng`. NumPy and IPython were also upgrade and require changes to Sage. This was discovered during the review of #221634.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build sage

**Comment by wegank** [comment]
Looks like sage still has doctests failing on master.

@amesgen, would you like to take a look?

**Comment by collares** [comment]
I am fixing this. Thanks!

File: pkgs/applications/science/math/sage/sagelib.nix [hunk 1]
```diff
@@ -119,6 +120,7 @@ buildPythonPackage rec {
     givaro
     glpk
     gsl
+    libpng
```
Comment (collares) [inline pkgs/applications/science/math/sage/sagelib.nix:123]: Should this be in `propagatedBuildInputs`? Tests also seem to pass changing `buildInputs` instead.

Comment (wegank) [reply pkgs/applications/science/math/sage/sagelib.nix]: Let's give it a shot.

**Comment by collares** [comment]
Tests are OK. The aarch64-linux failure is a [previously-seen intermittent failure](https://github.com/sagemath/sage/issues/34926#issuecomment-1418193082), but I will file it as a separate issue upstream just in case.

**Comment by collares** [comment]
Since importing patches to fix staging-next breakage is fairly routine and the newly-imported patches are fairly minimal, I think I will just merge this PR to unbreak Sage. Feedback is still welcome and will be addressed in follow-up PRs (if needed).

**Comment by wegank** [comment]
Thank you!

**Pull request merged**